test/Sentry.Tests/Protocol/TransactionTests.cs

```diff
@@ -526,4 +562,70 @@ public void ISpan_GetTransaction_FromSpan()
         // Assert
         Assert.Same(transaction, result);
     }
+
+    [Fact]
+    public async Task Idle_transaction_should_finish_within_idle_timeout_specified()
+    {
+        // Arrange
+        var client = Substitute.For<ISentryClient>();
+        var options = new SentryOptions
+        {
+            Dsn = ValidDsn,
+            IdleTimeout = TimeSpan.FromMilliseconds(2)
+        };
+        var hub = new Hub(options, client);
+        var context = new TransactionContext(
+            SpanId.Create(),
+            SpanId.Create(),
+            SentryId.Create(),
+            "my name",
+            "my operation",
+            "description",
+            SpanStatus.Ok,
+            null,
+            true,
+            TransactionNameSource.Component
+        );
+
+        var transaction = new TransactionTracer(hub, context);
+
+        // Act
+        await Task.Delay(20);
```

Comment:
@SeanFeldman is there any reason to use `await` here? Isn't the goal to make sure that the timer has had a chance to expire **_before_** continuing with the check in your test? You don't want this happening in the background right?

Maybe just `Thread.Sleep(20)` instead.